### ISSUE TITLE
Q: Family synonyms when a family is divided.

### ISSUE THREAD

**Issue by RoDuth** [Ghini/ghini.desktop#79]
Came across this issue when cleaning up a few name changes from the latest Queensland Herbarium records and I'm not sure of the best way forward. @tmyersdn Maybe you have an approach to this?

_Euphorbiaceae_ is one example.  Now we have _Pittosporaceae, Picrodendraceae_, etc. and _Euphorbiaceae_ also still exists.  I'm adding some of these new families and splitting out the genera and am trying to work out how to best capture synonyms.  If I have a bunch of genera (e.g. _Dissiliaria_) which have been moved from _Euphorbiaceae_ to _Picrodendraceae_ etc..  I was hoping that you would be able to create these other families, list _Euphorbiaceae_ as a synonym then just change the Family in the Genus Editor for each genus, job done.  Problem is Families are only allowed to have the one accepted name it seems.  So when you try to add _Euphorbiaceae_ as a synonym of both _Pittosporaceae_ and _Picrodendraceae_ Ghini displays an error when you try to add it to the second family.  The only way I can see of making it work would be to: 
1. Create the new families, 
2. Change the family in the Genus Editor for each genus (therefore taking along all the attached species)
3. Create a new genus for each one that has been moved with the old family (_Euphorbiaceae_) and same genus name again.
4. Go back and edit the Genus in the new Family again adding the _Euphorbiaceae_ genus as a synonym.

Thats a lot of work!  Anyone got a better method?  And _Sterculiaceae_ is in a mess also!

@mfrasca Is it possible to make it that when you change the Family name in the Genus Editor a helper automatically pops up (like the one that shows the plantlist info when you click the tick button in the species editor etc.) asking "do you wan to keep _Dissiliaria_ (_Euphorbiaceae_) as a synonym?" with yes/no buttons that would complete steps 3 and 4 above for you?

**Comment by RoDuth** [comment]
And at species level too!!  Just had to do the 4 step thing for a species name change....  Change the name and save, create the old species again and save, edit the new name again and add the synonym and save.

**Comment by RoDuth** [comment]
And just noticed that the Plantlist name checker (loving this feature :+1: ) doesn't correct the family if it's wrong even though it does display it.

**Comment by mfrasca** [comment]
when splitting families this way, keeping the original name for a smaller set of genera and putting most of the genera into new families, I guess this is a case where we would use the **aggregate** flag.
- the old family, in your case _Euphorbiaceae_, would be come an aggregate family, initially with all genera referring to it sensu lato. if "sensu lato" were the default, you would not need to much here.
- you would correct the genera which stay in the original family as referring to it sensu stricto.
- you would create new families and state they have as synonym the original one sensu lato. this feels to me as forcing the concept of synonym into something else.
- you would move the genera to their correct family.

some of the above steps are not supported.

we would end with a radically different representation of reality in the database, and I am not sure the one I'm suggesting would solve all issues you think you will encounter.

**Comment by tmyersdn** [comment]
Hi @RoDuth and @mfrasca 

Ross and Mario, you are teaching me some botany.

Mario, it certainly does look like 'sensu stricto' and 'sensu lato' are used in relation to families:
https://en.wiktionary.org/wiki/sensu_stricto#Latin
https://en.wiktionary.org/wiki/sensu_lato

I can see why you suggest the aggregate flag, and if you search on Google, you can find examples of "aggregate families", eg the Malvaceae. However, I think this is a very academic matter, and not very practical for small regional garden to keep up to date with, when in reality, they just want to update to the correct family name to make good labels.


Ross, we have no short cut on our database, we have to manually go through each genus and update the family.  But, I definitely agree with you that each family should be able to have more than one synonym, as so many plants have been through several family changes. Would use of a multiple value field fix this?

I am sorry, I have not had time to try this on Ghini, but can I ask how your step (3) is helping you? Will this help you to find the plants that need their family changed, or are you trying to keep track of the plants that used to belong to Euphorbiaceae, in case they change back again? I don't see how it helps to have a second genus entry with the old name. Sorry this is probably obvious within Ghini!

I am keen to see ThePlantList checker and how it presents family names. I will have more questions about that. My immediate thought is that when Ghini finds a family change on ThePlantList, that this gets added to a standard tag (eg "Family change"), then the user can decide whether to make the change or not. Of course it would help if there was a tool to effect the change.

Finally for your interest, have you seen this list of families and what different experts call them:
http://www.plantsystematics.org/reveal/pbio/usda/usdaindex.html

**Comment by RoDuth** [comment]
Wow its been way too long since I last checked in! :smile: 

@tmyersdn Yes, I just want to be able to see that _Pittosporaceae_ was once consider _Euphorbiaceae_ and I don't think the aggregate family is the best way to deal with the issue.  Also, its not that a family can't have more than one synonym, it can in Ghini.  Its that a family can only be the synonym of 1 other family.  Which isn't going to work with a family that has been divided into several others, as in the _Euphorbiaceae_ example I gave.  But maybe this isn't really that great an issue...

The greater issue is at species level. i.e. in order to not lose all associated data you have to do the steps I describe above. Open species and rename it to the new name - recreate the original species name - edit the renamed species - add the recreated old species name as a synonym of the renamed species... if that made any sense??!! try it and you'll see what I mean. Just try to take a species with a lot of accessions, notes, etc. and change its name without losing any data but keep the old name as a synonym.  For instance, we had Mackay Cedar in a lot of places in the gardens and it had a name change from _Paraserianthus toona_ to _Falcataria toona_ not too long ago.  This will be recorded as _Paraserianthus toona_ in lots of the old documents.  I want to be able to put _Paraserianthus toona_ into Ghini and see that its had a name change.  The only way I can see is the convoluted process I'm talking about.  The process went like this: create the genus _Falcataria_ - open _Paraserianthus toona_ and change the genus field to _Falcataria_ close this saving changes - create a new species entry for _Paraserianthus toona_ close this then reopen the recently changed entry for _Falcataria toona_ and add _Paraserianthus toona_ as a synonym in the "Additional info" tab.    Not much fun when half the members of a genus have had their genus changed to something else (e.g. when Qld Herbarium decided that they believed all Callistemon are now Melalueca but all the other herbariums still use Callistemon.  We changed all the Queensland species over to Melalueca but not the ones from elsewhere and we wanted to be able to search for  _Callistemon viminalis_, the name we had used, and see that it is now _Melaleuca viminalis_. )  It was a lot of work and it needn't be in my opinion, especially for something so common and core to the purpose of the database.

@mfrasca I just noticed that the name checking function tells you if the name you have is a synonym and asks of you would like to add the correct species name to the database.  It then adds the species and marks the current name as a synonym, which is all fine but this means that you have to manually change the species name for each accession because they are all left behind still attached to the old (wrong) name.  So I'm assuming that in your version of dealing with a synonym you do it differently to what I've suggested. i.e. you create the new correct name and mark the old name as a synonym then open each of the accessions that are attached to it and change the species?  But then what about any attached data (e.g. notes, pictures, vernacular names, distribution) this would all remain attached to the now wrong name wouldn't it and you would have to copy it across or something?

**Comment by mfrasca** [comment]
@RoDuth , wait, until we have reorganized the database structure as of "issue 9" (now moved to #92 ), we still handle family/genus/species separately. as of now, only for species there's the possibility to associate accessions to the record.

* I would not use the term "wrong" for names that are considered synonyms.
* I agree that when you are looking at an object that is considered a synonym or has one or more synonyms, it would be useful if the software would inform about all objects that are related through the other taxa. like: you have _Iris_ ×_florentina_ & _Iris_ ×_germanica_, and the first is considered synonym of the second. but accessions associated to one taxon will not be shown nor reported in any way in a search for the other. this might be a problem and you might want to report it.

I still have to read most of your comment, this is a partial response.

**Comment by mfrasca** [comment]
one thing I miss in the above comments, it's something I have done in the ghini.pocket and ghini.web databases, and which can't be done in ghini.desktop, at least not until we solve #92 : include subfamilies and allow for inter-rank synonymies.

in ghini.pocket the database allows for derivations like:
genus: Theobroma, L.
family: Sterculiaceae, Vent. (synonym of the following)
subfamily: Sterculioideae, Beilschmied
family: Malvaceae s.l., Jussieu
order: Malvales, Berchtold & J. Presl

without a properly defined database, you cannot get it right.  just imagine, even Tropicos seems to have this mistake, separating taxa per rank, thus preventing inter-rank synonymies:
http://www.tropicos.org/Name/42000102
(from Sterculiaceae, straight via a synonymy to Malvaceae, which is not exactly the case.  the case is a combination of synonymy and inclusion, as illustrated above.)

@RoDuth , your Euphorbiaceae example, there taxonomists did the opposite of what was done for Malvaceae (splitting vs. grouping)… `Now we have Pittosporaceae, Picrodendraceae, etc`.  Was there (I mean before splitting the family) anything like subfamilies or anyhow infra-familiar taxa, elevated to rank family?

@tmyersdn , `they just want to update to the correct family name to make good labels.` … I've seen, in smallish gardens, the effort to include subfamily information on Malvaceae labels.  apparently their conservators felt the need to keep a memory of the older names, but show they are up to date.  what I saw did not include the `s.l.` indication.  so the information on the labels I've seen was: binomial → subfamily → family. (e.g.: Theobroma cacao → Sterculoideae → Malvaceae)

an other example from ghini.pocket:
genus: Malva, L.
family: Malvaceae s.s., Jussieu (synonym of the following)
subfamily: Malvoideae, Burnett
family: Malvaceae s.l., Jussieu
order: Malvales, Berchtold & J. Presl

please note that this database has two separate entities, one for Malvaceae s.l. and one for Malvaceae s.s..  I'm not convinced this is a good idea, but I didn't see how to do it otherwise.

@tmyersdn, I guess you would rather skip the Malvaceae ss/Malvoideae steps, and do without the `s.l.`indication.  leave it to an algorithm, and you get Malvoideae on your labels.

**Comment by tmyersdn** [comment]
Hi @mfrasca, 

thanks for raising this question - this is a good point, I think all gardens may want the flexibility to add subfamily information onto labels. In Dunedin we do this for rhododendron species (subection name or sometimes section name): **https://en.wikipedia.org/wiki/List_of_sections_in_subgenus_Rhododendron**
But also for cultivar group names for roses and camellias.  In general, we place this information in a different part of the label to the family name.

I am curious, did you resolve the first question above? "when you try to add Euphorbiaceae as a synonym of both Pittosporaceae and Picrodendraceae Ghini displays an error when you try to add it to the second family"? Does it work to use the aggregate flag?

**Comment by mfrasca** [comment]
oh, I'm sorry, apparently I got lost in the discussion and missed the point.

»when you try to add Euphorbiaceae as a synonym of both Pittosporaceae and Picrodendraceae Ghini displays an error when you try to add it to the second family.«

I have tried this, and I observe that you can do it just fine,**given that all three families are already present in the database**.

I observed that Picrodendraceae is **not** in our initial database, so you indeed see an error when you try to add **this** second family, but not if you try to add just any family.   if you type slowly, you see the list of hints and see that it does not contain the desired epithet.  the reason for this is not that »Families are only allowed to have the one accepted name« (I guess there is a mistake in this statement, and that @RoDuth meant it the other way round: Families are only allowed to have the one synonym), but anyhow, neither of the two statement is the problem here.  the problem was that Picrodendraceae was not in the database.

add it, and you can select it as a synonym for Euphorbiaceae.

**Comment by mfrasca** [comment]
@tmyersdn »Ledum […] is now recognised as a subsection of section Rhododendron, subgenus Rhododendron, of the genus Rhododendron.« what's the polite way to exclaim "shit!" ??
I'm not sure how I would address this, I've always seen taxa where the epithet hinted at the rank.  but never something like genus, subgenus, section, all the same epithet, and what about authorship??
anyhow, I'll have a look.
do you have more reading material on the subject?

**Comment by tmyersdn** [comment]
ahh, doesn't this happen a bit when taxa are merged due to taxonomic revision -  eg look at what happens to Michelia (**https://en.wikipedia.org/wiki/Magnolia**)

As a separate issue, looking at Ledum, it is interesting to see the difference between ThePlantList and GBIF in terms of adopting name. See the table at the end of this page: **https://en.wikipedia.org/wiki/Rhododendron_subsect._Ledum** NB: this needs verifying..

**Comment by mfrasca** [comment]
Well, the annoying part with this way of reorganizing (moving a taxon (e.g.: Ledum) from one point to an other in the taxonomic ancestry tree) is that the reorganized taxa are still called precisely the same.
With Sterculiaceae, they added a new name Sterculioideae, defined the synonymy, and stated that Sterculioideae (clearly from its epithet a subfamily) is a subfamily of Malvaceae s.l.
But Sterculiaceae is still a famiiy, and it is still within the Malvales order, it's just not advised to keep using it in newer publications, this is what I understand.
sorry, by "more reading" I actually meant if you have the complete current derivation tree from genus Rhododendron down to all its subsections.
this is the first time I meet these ranks (subgenus, section, subsection).

**Comment by RoDuth** [comment]
Can I just bring this back to the original reason for this issue.  It was meant to be more about workflow when changing a taxon name (see my comment in the original issue about adding a pop up to complete steps 3 and 4).  Included in this is the desire to be able to have more than one taxon having another single taxon as a synonym.  (i.e. Pittosporaceae, Picrodendraceae should both be able to have Euphorbiaceae as a synonym).  Anything else is just a complication.  I've said before that it is rare that I have the need for anything like subfamily, tribe or any of the other ranks (subtribe, section, series, etc.) and I have no desire to try to keep those extra parts up to date.  We never use them on labels and hardly ever use them in fact sheets and the like (looking it up at the time when we do).  If you do include these extra ranks then you must include all the common ones ([see Wikipedia article](https://en.wikipedia.org/wiki/Taxonomic_rank#Ranks_in_botany)).  And the name checker needs to be able to check and correct these also.

A bigger issue for workflow for synonyms is in this example I gave for a simple species name change:

> Just had to do the 4 step thing for a species name change.... Change the name and save, create the old species again and save, edit the new name again and add the synonym and save.

See also https://github.com/Ghini/ghini.desktop/issues/79#issuecomment-228956622. This should be able to be automated.

**Comment by RoDuth** [comment]
if you are looking for other examples of renaming that are a bit of a mess regarding rank look at [Fabaceae](https://en.wikipedia.org/wiki/Fabaceae), It used to be Mimosaceae, Fabaceae, Caesalpinaceae when I first started at this and now its Fababaceae as the family, there are now 6 sub families with one Caesalpinioideae being the old Caesalpinaceae while Mimosaceae ends up as Mimosoideae, a [clade](https://en.wikipedia.org/wiki/Clade) below Caesalpinioideae.  Are you going to allow for clades also?  Also don't forget the [APG IV system](https://en.wikipedia.org/wiki/APG_IV_system)

All seems like a hell of a lot of work to me best left to the flat dead plant collectors (herbarium botanists).  I just want to be able to move all my Mimosaceae, Caesalpinaceae into Fabaceae and make them synonyms of Fababceae so can I find them in the future.

**Comment by mfrasca** [comment]
On 10/04/2018 10:13 PM, RoDuth wrote:
> Pittosporaceae, Picrodendraceae should both be able to be synonyms of
> Euphorbiaceae
and you have any difficulty in achieving this?
did you make sure all three taxa are in the Family table, before trying to lay the connection?

**Comment by mfrasca** [comment]
On 10/04/2018 10:30 PM, RoDuth wrote:
>
> All seems like a hell of a lot of work to me best left to the flat
> dead plant collectors (herbarium botanists). I just want to be able to
> move all my Mimosaceae, Caesalpinaceae into Fabaceae and make them
> synonyms of Fababceae so can find them infuture.
>
with the future "Taxonomy" plugin replacing the current "Plants" plugin you do not need much to correct the situation, really hardly anything.
e.g.: you add a new taxon Mimosoideae, as subfamily of Fabaceae, and state that Mimosoideae is the accepted taxon for Mimosaceae. you do this for all the "old" families, and you're ready to go, according to me.
when printing the taxonomic derivation of, say, Senna, you skip the non-accepted family (in this case, Cesalpinaceae), you print the accepted subfamily (in this case, Cesalplinoideae), and then you tell me if you really need anything else.
you ask me if I want to do Clades?  the amount of detail you want, you decide yourself, and what you put it in the database, that's what you get back from the program.  if you don't need Clades, you don't put Clades.
The point about the Taxonomy plugin is really: who's financing it?  I'm putting tons of my time and money, and I'm interested in programming it, but I wish that someone else showed some interest, too.

**Comment by mfrasca** [comment]
On 10/04/2018 10:13 PM, RoDuth wrote:
>
> A bigger issue for workflow for synonyms is in this example I gave for
> a simple species name change:
>
>     Just had to do the 4 step thing for a species name change....
>     Change the name and save, create the old species again and save,
>     edit the new name again and add the synonym and save.
>
> See also #79 (comment)
> <https://github.com/Ghini/ghini.desktop/issues/79#issuecomment-228956622>.
> This should be able to be automated.
>
what about you describe it as if it already existed?  like writing the user manual page relative to this functionality.  then I take care to make sense of what you wrote, possibly rewrite it but maintaining the underlying functionality, then realize what the user manual describes. as of now, I would not really know where to start nor in which direction to go.

**Comment by mfrasca** [comment]
> interesting to see the difference between ThePlantList and GBIF in terms of adopting name

some time ago (May 2015) I had the chance to see a partial export from the MoBot database, limited to Ecuador Orchids.  they represent each synonymy not as a **fact**, rather as an **opinion**.  and each opinion is associated with the authorities that support it.  their database holds (all) known alternative opinions.  I don't think I would like to do this in ghini.

**Comment by mfrasca** [comment]
I am currently thinking of associating each rank to a floating point number that helps putting ranks in order.  
an **example** could be: from Regnum (0.0) to Ordo (4.0), Familia (5.0), Genus (7.0), Species (9.0), you could add (individually named) ranks Subgenus (7.5), Section (7.75), Subsection (7.9).  then if you think so you add (unnamed ranks, i.e.: using the generic 'Clade" name) at intermediate levels.
that is, you would use these values if you think they would match your needs.
when adding a new taxon, you choose the rank name (and implicitly its numeric depth), type the epithet and authorship, then choose the parent taxon in the hierarchy, among the taxa with strictly lower rank value.  and for synonymy, you would first choose the rank of the accepted taxon, then choose the epithet.
this a working option, and I'm not sure it would work in practice.  I guess it would.
for clades, the interface should be expanded, but I guess that the database structure would allow for them.

**Comment by mfrasca** [comment]
> See the table at the end of this page: https://en.wikipedia.org/wiki/Rhododendron_subsect._Ledum NB: this needs verifying..

interesting.  in my opinion the table would be more informative if the two columns GBIF Status / TPL Status (now only holding the text "Accepted", or empty) would hold the accepted status of the GBIF/TPL accepted species according to the TPL/GBIF database respectively.  but then again, this only stresses the fact that "Acceptedness" is an opinion.  ;-)

**Comment by mfrasca** [comment]
@tmyersdn ,  about the Magnolia example, I had a moment of panic…
![image](https://user-images.githubusercontent.com/122869/46574106-21749080-c964-11e8-94a6-b379846c94a9.png)
but it does make sense after all.
when you add, say ›Magnolia globosa‹ to the database, you don't add it straight into the ›Magnolia‹ genus, you add it to the ›Oyama‹ subsection, with derivation Section ›Rhytidospermum‹, Subgenus ›Magnolia‹.  its binomial name however is composed of Genus-Species epithets.

when you have infrageneric information, the software should warn against inserting a species straight into the genus. (I'm describing the new, not existing, Taxonomy plugin).

the remaining question, I don't know if you are able to answer it, is  how would you indicate an incomplete verification, at infrageneric rank?  say you have an accession and you know more than just "it's a Magnolia", like you know it belongs to a species in the ›Elmerrillia‹ subsection.  how do you indicate this?  my guess (just a guess): **_Magnolia_ subsect. _Elmerrillia_ sp**

**Comment by RoDuth** [comment]
> Pittosporaceae, Picrodendraceae should both be able to be synonyms of Euphorbiaceae

I had corrected that to read
> Pittosporaceae, Picrodendraceae should both be able to have Euphorbiaceae as a synonym

This I can not do.
You must have read it before I edited sorry.

As for Fabaceae.  I'm not sure I would say that Caesalpinioideae is the accepted name for Cesalpinaceae as Caesalpinioideae now also contains the Mimosoideae clade which contains the previous members of Mimosaceae.  I would rather just leave it at family level.

**Comment by mfrasca** [comment]
> On 10/08/2018 03:09 PM, RoDuth wrote:
> Pittosporaceae, Picrodendraceae should both be able to have
> Euphorbiaceae as a synonym
>
> This I can not do.

no, you cannot.

two taxa A and B having both a third taxon C as a synonym would mean that that both taxon A and taxon B are the accepted taxon for taxon C.  this information you cannot put in the database, no.

I wouldn't be able to show you this in the database structure (Plants plugin) because it's unnecessarily quite convoluted according to me.  I've ignored how it was programmed and rewrote it from scratch in the new Taxonomy plugin.  it is now one single foreign key, `taxon.accepted_id` → `taxon.id` .

but I also have trouble following the example you're making.  unless you are missing the "s.l." after Euphorbiaceae, that is.  are you maybe intending to register both families as lower-rank taxa of the
"superfamily" Euphorbiaceae s.l.?  something like what's written in the wikipedia page for this section in the Asparagales structure?

![image](https://user-images.githubusercontent.com/122869/46633516-799dc500-cb13-11e8-8605-b17edb0409b8.png)

**Comment by mfrasca** [comment]
I have been quicky-reading about this continuously changing mess, where taxonomists are trying to mirror the most recent DNA insights in the plant taxonomy.  in practice what I now see is that the very concept of rank is fading away, in favour of clade, group, and what-not.
how to represent this in the current database structure (1.0; 3.1)?  you don't, not with ghini's help and I guess not at all.  we only have family-genus-species, notes, and no inter-rank synonymies.
how to do this in the new `taxonomy` plugin I'm playing with … I guess the way would be that you switch off the check "including taxon must be at higher rank".  then you would be able to define the ranks "Clade" and "Group" and use them wherever you like, if you like.
again, the question if one would want to follow the flow and do this, it's up to you users.

**Comment by tmyersdn** [comment]
@mfrasca, this sounds good.  Also, I think it is important to look at where online data can come from. I see wikipedia sometimes offers clade information, sometimes traditional taxon ranks.  Presumably they take their data from EOL.
Access to TPL data in Ghini is great (it is amazing to see this working), but maybe, EOL or GBIF might give more fluid insight to subfamily taxon groups (if you think that is important).
I think you are right, and that this discussion needs to be addressed by an actual user of Ghini.  We certainly have historical garden plantings that are grouped by old taxonomy: eg a collection of azaleas.
great to see the discussion above :)

**Comment by RoDuth** [comment]
I realise synonym is most probably not the best way of saying it but I want a way to state they were previously part of Euphorbiaceae.

> but I also have trouble following the example you're making.  unless you are missing the "s.l." after Euphorbiaceae, that is.  are you maybe intending to register both families as lower-rank taxa of the "superfamily" Euphorbiaceae s.l.?  something like what's written in the wikipedia page for this section in the Asparagales structure?

definitely not, I'm just trying to record the splitting up of Euphorbiaceae.  (currently I'm using notes and I'm happy to stick with that for now)

> in practice what I now see is that the very concept of rank is fading away, in favour of clade, group, and what-not.

This is why I mentioned  clades and APG IV earlier (there are still plenty that use [conquist](https://en.wikipedia.org/wiki/Cronquist_system)).  I have no need to get into all that.  I notice that the main sources I use for names almost never go into more detail than family.  (Atlas of living Australia and Queensland Herbarium).  The binomial is the key the rest is best left to the taxonomists to argue about.  I mean some still insist on using Leguminosae!  ...I see ABCD and DarwinCore still use the traditional ranks as you have included them.

**Comment by RoDuth** [comment]
Can I add, if there is any obviously missing taxonomic elements (for things such as orchidaceae) its the grex name and the intergeneric hybrid flag. (see `genus where genus like 'x %'`)
e.g. Paphiopedilum Jim Kie 'Springwater'.  The other is a decent way to  handle provisory names...  (I have no idea where your Zzz names have come from or why you use them.)  Here are 2 examples: [Gen. (Aq520454) sp. Shute Harbour (D.A.Halford Q811)](https://bie.ala.org.au/species/http://id.biodiversity.org.au/node/apni/2905306#classification), [Argyrodendron sp. Whitsundays (W.J.McDonald+ 5831)](https://bie.ala.org.au/species/http://id.biodiversity.org.au/node/apni/2891976)

**Comment by mfrasca** [comment]
Grex? Group? #260 #259 
Group is possibly included already in what I'm doing for 3.2.
The funny idea of Grex I would not yet know, it looks like something **next** the the derivation tree.

hybrids, I plan to make them in 3.2 as I had briefly approached them in 1.1.  that is: a taxon (at whatever rank as far as the database is concerned) can be specified by its epithet+author+year, or by a hybrid formula, where you link it to two (or more) other taxa at the same rank.  if one of the referred taxa is itself a hybrid, it will appear in parenthesis in the formula.  if the taxon is specified by an epithet, it can still be a hybrid or a graft chimera, and you indicate it by setting a flag.  The query would possibly look like: `genus where hybrid is True` or `genus where hybrid_flag = 'H'`.  I have to look into this again, I've nothing else than implemented the ITF2 recommendations.

--------------------------------------------------------------
The `Zzz-` prefix come from practice, in collections where plants have been **not** better identified than rank Family.  

Bauble and Ghini up to 3.1 need to identify an accession to rank Species or better.  which raises the need for fictive species.  if the identification reaches rank genus, you simply add the fictive species `<genus.epithet> sp.`, if it does reaches rank Family, you first add a fictive genus `Zzz-<family.epithet>`, then the fictive species `Zzz-<family.epithet> sp.`.
The triple Z is only a trick to make sure they are at the end of the list.  I could have used a triple A but I found that confusing, in particular considering the existence of genus Aa.
Things get a lot more complex in Orchidaceae collections, where I have met cases identified to rank Tribus, or Subtribus, and obviously we have no such things in the database.  you don't want to add a `Zzz-<tribus.epithet>` because this way there would be no simple way to select all such accessions.  

But this is a different discussion and I have very recently opened it on the forum and there was very little response.  You may revive it there, if you think so.  https://groups.google.com/forum/#!topic/bauble/IV6oyK01a5o

----------------------------------------------------
your more complex example:
Gen. (Aq520454) sp. Shute Harbour (D.A.Halford Q811)
…
I can make 3.2 do this:

        asterales = Taxon(rank=self.ordo, parent=self.plantae, epithet='Asterales')
        asteraceae = Taxon(rank=self.familia, parent=asterales, epithet='Asteraceae')
        gen_nov = Taxon(rank=self.genus, parent=asteraceae, nov='Aq520454')
        sp_nov = Taxon(rank=self.species, parent=gen_nov, epithet='Shute Harbour', nov='D.A.Halford Q811')
        self.assertEquals(gen_nov.show(), 'Gen. (Aq520454) sp.')
        self.assertEquals(sp_nov.show(), 'Gen. (Aq520454) sp. Shute Harbour (D.A.Halford Q811)')

notice the trailing `sp.` when identifying an accession to rank genus.

but I doubt about the `Shute Harbour`, which looks like a cultivar name.  are you sure you Australians want it like you're reporting it?  can you live (better) with: `Gen. (Aq520454) sp. (D.A.Halford Q811) 'Shute Harbour'`?

**Comment by RoDuth** [comment]
As for the work flow issue:

re:
> A bigger issue for workflow for synonyms is in this example I gave for a simple species name change:
    Just had to do the 4 step thing for a species name change.... Change the name and save, create the old species again and save, edit the new name again and add the synonym and save.
See also #79 (comment). This should be able to be automated.

and 
> as of now, I would not really know where to start nor in which direction to go.

Possible solution:
When you edit any of the taxonomic fields for a family, genus, species and click ok you have a pop up that gives you the option to "create copy of original as a synonym" or "edit original only".  Similar to the way that you currently get a "Are you sure you want to lose your changes?" if you click cancel.  If you select "create copy of original as a synonym" a new entry is created with the old name and it is added as a synonym for the new taxon just saved.

Have you tried to change a species name lately?  For something that has a lot of data (e.g. notes, pictures, attached accessions, etc.), __without__ losing any of that data. When Qld Herbarium merged Callistemon into Melaleuca we had to do this 4 step process (from my comment [above](https://github.com/Ghini/ghini.desktop/issues/79#issuecomment-228956622)) for something like 30 taxa in Callistemon.

> Open species and rename it to the new name - recreate the original species name - edit the renamed species - add the recreated old species name as a synonym of the renamed species..

Very laborious.

**Comment by mfrasca** [comment]
> I realise synonym is most probably not the best way of saying it but I want a way to state they were previously part of Euphorbiaceae.

I don't know how to say this, but (1) being synonyms, and (2) being part of, that's two quite different concepts.  if you want to record A and B as part of C, you use taxonomic inclusion, which you say you "definitely not" want to do.  statement that leaves me puzzled.  anyhow, that you do or do not want to do this, it doesn't make much of a difference, because bauble and ghini up to 3.1 do not offer any flexibility in taxonomic inclusion, only species within genus within family.  full stop.  so as far as the original issue is concerned (**synonyms when dividing**), I think that the issue is caused by using brute force against a limitation I'm already busy removing.

You say you're solving this with notes, which I guess is your only available option.  it would be nice to see how you're organizing your notes for this, if you want to share your cases and solutions, possibly as "contributed recipe" in the docs.  it would be helpful so that I can consider your experience in 3.2.

further, there's a huge amount of different issues raised here, and I wish to keep them separated, or I'm unable to think straight (you've already noticed I'm not answering to all the questions being opened here).  would you please open separate issues for:

- joining two genera when they already have species with accessions.
- trouble changing a species name (I don't get the point, I'm afraid).
- representing unpublished taxa, and refer to them from an accession.
- how can a species refer to an unpublished genus?
- all other questions I've dropped.

also, please:
- (add your 2¢ to #260)
- (add your 2¢ to #259)

**Comment by tmyersdn** [comment]
@mfrasca & @RoDuth - hah, I am always too late with my comments, and then it is closed!

I think wikispecies has been trying to resolve some of these problems:
See: **https://species.wikimedia.org/wiki/Fabaceae**
(list of subfamily levels which can drill down to lower levels, then a break down of homotypic and heterotypic synonyms), also an interesting list of vernacular names at the end.
Certainly am not suggesting Wikipedia as the best place for taxonomy, but they certainly are actively dealing with public presentation of taxonomy at levels of genus and above, when many other sources are more cryptic.

**Comment by mfrasca** [comment]
@tmyersdn, don't worry on commenting anyway, it's an interesting discussion.
as an issue, I closed it because I do not see a sensible way to use the
available synonyms links to mean an unavailable inclusion link.  I made
a start programming the new taxon plugin for 3.2, but that's issue #92. 
we have mentioned a few other opportunities, related but not falling
under this same title, and I am interested though in seeing how @RoDuth
is using notes for taxon inclusion in 1.0, but I'm afraid I am repeating
myself.  ;-)

**Comment by mfrasca** [comment]
(what's that now … heterotypic and homotypic synonyms?  one more thing to learn …)
